### PR TITLE
Add version 1.1.0

### PR DESCRIPTION
**Pull request by lexming**
This repo provides the Python script used to convert documents in RST format from our website (news posts) to text files suitable to be visualized in a terminal as MOTD.

Main features:
* Custom translator for docutils to convert an RST file to text with ANSI escape codes
* HTML parser to identify RST files from HTML news feeds from ABlog
* Formatter of text files with ANSI escape codes: add header/footer, add extra link, wrap the text

The README has more information.

### CODE REVIEW

File: bin/news2motd [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+from term_rst_post.ansimotd import accomodate_motd, enable_motd
+from term_rst_post.exit import error_exit
+from term_rst_post.filetools import change_file_extension, direct_path, fileobj_extension, valid_dirpath
+from term_rst_post.filetools import common_path_join, rst_path_from_html_link
+from term_rst_post.newsfeed import get_top_ablog_news, valid_url
+from term_rst_post.newspost import get_post_info_from_rst, make_ansicode_from_rst
+from term_rst_post.version import VERSION
+
+DEFAULT_LIFETIME = 48
```
Comment (wpoely86) [inline bin/news2motd:50]: please add a comment with the unit :wink:

File: bin/news2motd [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        except (AttributeError, FileNotFoundError) as err:
+            error_exit(err)
+    else:
+        try:
+            # Parse RST and get MOTD date
+            motd = get_post_info_from_rst(args.newspost)
+        except (IndexError, ValueError) as err:
+            error_exit(err)
+
+    # Geerate link to HTML page
```
Comment (wpoely86) [inline bin/news2motd]:
```suggestion
    # Generate link to HTML page
```

File: src/term_rst_post/ansimotd.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+import re
+
+from textwrap import indent, wrap
+
+from term_rst_post.exit import error_exit
+
+logger = logging.getLogger()
+
+
+def accomodate_motd(motd_dir, body_file, head_file=None, foot_file=None, foot_link=None, wrap_width=70):
```
Comment (wpoely86) [inline src/term_rst_post/ansimotd.py]: `motd_dir` is unused? Or I don't see it ?

File: src/term_rst_post/ansimotd.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        logger.info(f"Reformatted contents of ANSI text file '{body_file.name}' to {wrap_width} characters")
+        motd_file.close()
+
+    return motd_file
+
+
+def enable_motd(motd_cfg, motd_file):
+    """
+    Enable MOTD message
+    - motd_cfg: (string) path to config file in MOTD directory
```
Comment (wpoely86) [inline src/term_rst_post/ansimotd.py]: is the motd_cfg still relevant with this?

File: src/term_rst_post/filetools.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        new_filepath = new_filepath.name
+
+    logger.debug(f"File path '{new_filepath}' with extension '{extension}' generated from path '{filepath}'")
+
+    return new_filepath
+
+
+def direct_path(rel_path):
+    """
+    Return path without special maps
```
Comment (wpoely86) [inline src/term_rst_post/filetools.py]: doesn't python have a function for that? abs path or so?

**Review by wpoely86 — COMMENTED** [review]
I've done a first scroll through but as it's kinda big I think I need another proper look.

File: bin/news2motd (hunk 1 above)
Comment (lexming) [reply bin/news2motd:50]: sure https://github.com/sisc-hpc/term-rst-post/pull/1/commits/7b639a010b4c135a273d5b253e39f7de15662915

File: bin/news2motd (hunk 2 above)
Comment (lexming) [reply bin/news2motd]: https://github.com/sisc-hpc/term-rst-post/pull/1/commits/be420c4ca1746088dec73d002d58f7ccf17b41c6

File: src/term_rst_post/ansimotd.py (hunk 4 above)
Comment (lexming) [reply src/term_rst_post/ansimotd.py]: very good point, given how the Azure pipeline works this is indeed unnecessary. I removed it in https://github.com/sisc-hpc/term-rst-post/pull/1/commits/6832337c17938e38276f71f66dd8500e6c10f91c. Now `news2motd` generates a file with a known name, which can be controlled with `--motd-path`.

File: src/term_rst_post/ansimotd.py (hunk 3 above)
Comment (lexming) [reply src/term_rst_post/ansimotd.py]: Indeed, it was a leftover from https://github.com/sisc-hpc/term-rst-post/pull/1/commits/5f67a0313acd4378f7aa249eb640d2d29c5ad6f0. Now it is totally removed in https://github.com/sisc-hpc/term-rst-post/pull/1/commits/afc8ec7bb7c1404633c7e7b16e9dcbad68f37944

File: src/term_rst_post/filetools.py (hunk 5 above)
Comment (lexming) [reply src/term_rst_post/filetools.py]: this function is actually quite dumb, the goal is just to remove any leading special maps from the path, not to resolve the path. I updated the function renaming it to `inward_path` to make it more clear. You can check the test `test_inward_path` in `test_filetools.py` to see some examples of expected results.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# Byte-compiled / optimized / DLL files
```
Comment (wpoely86) [inline .gitignore:1]: can you add vim files?
```
# vim
*.sw[op]
*~
```

File: bin/news2motd [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+# Copyright 2021-2021 Vrije Universiteit Brussel
+#
+# This file is part of term_rst_post,
+# originally created by the HPC team of Vrije Universiteit Brussel (https://hpc.vub.be),
+# with support of Vrije Universiteit Brussel (https://www.vub.be),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/sisc-hpc/term_rst_post
```
Comment (wpoely86) [inline bin/news2motd]: do a search and replace for `sisc` :wink:

File: README.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  * emphasis: ANSI underline text
+  * literal: markdown literal plus ANSI inverse text
+  * reference: markdown links
+  * bullet list: markdown unordered list
+  * enumerated list: markdown unordered list
+* Support for custom substitutions:
+  * ``|Warning|``: ANSI red background
+  * ``|Info|``: ANSI green background
+  * other substitutions will be converted to plain text
+* Support for ABlog's update directive
```
Comment (wpoely86) [inline README.md]: maybe add a link to ABlog?

File: src/term_rst_post/ansimotd.py [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+    return motd_file
+
+
+def wrap_ansicode(ansistr, column_width, match_width):
+    """
+    Cuts long string to column width accounting for ANSI escape code
+    - ansistr: (str) text with ANSI escape codes
+    - column_width: (int) charcater length of column
+    - match_width: (int) number of charcaters to match between text representation
```
Comment (wpoely86) [inline src/term_rst_post/ansimotd.py]: I'm not sure I understand what this means? also 'charcater' :wink:

File: .gitignore (hunk 6 above)
Comment (lexming) [reply .gitignore:1]: Done in https://github.com/vub-hpc/term-rst-post/pull/1/commits/2f36fb614a5f5dcf662804d2ba90c1bd0bab2066

File: bin/news2motd (hunk 7 above)
Comment (lexming) [reply bin/news2motd]: Done in https://github.com/vub-hpc/term-rst-post/pull/1/commits/aa2bb48a0d06c26369b3d9575f8874b1ef80056b

File: README.md (hunk 8 above)
Comment (lexming) [reply README.md]: Done in https://github.com/vub-hpc/term-rst-post/pull/1/commits/314e43e833b383e8670aa34b116b68f90dfa87e6

File: src/term_rst_post/ansimotd.py (hunk 9 above)
Comment (lexming) [reply src/term_rst_post/ansimotd.py]: Wrapping the rich ansi text works by using a wrapped non-ansi version as reference. `match_width` defines the length of text around the end of line of the wrapped non-ansi text that will be used to find the equivalent point in the rich ansi text and wrapped it. I made this argument optional as it is a technical parameter and also expanded the various comments in the function in https://github.com/vub-hpc/term-rst-post/pull/1/commits/78bd61e93cca27668c1b436704d4f35dfddee163

**Review by wpoely86 — APPROVED** [review]

**Pull request merged**